src/crab/splitdbm/adapt_sgraph.hpp

```
@@ -27,87 +28,61 @@ class TreeSMap final {
     };
 
-    struct KeyConstRange {
+    struct KeyConstRange : std::ranges::view_interface<KeyConstRange> {
         using iterator = KeyIterator;
 
-        explicit KeyConstRange(const col& c) : c{c} {}
-        [[nodiscard]]
-        size_t size() const {
-            return c.size();
-        }
+        KeyConstRange() : c{&empty_col()} {}
```

Clang (14.0.6)

[warning] use '= default' to define a trivial default constructor

(modernize-use-equals-default)